src/core/tests/Test_ConfigurePatchingProcessor.py

```diff
@@ -141,7 +141,12 @@ def test_operation_fail_for_configure_patching_agent_incompatible(self):
             substatus_file_data = json.load(file_handle)[0]["status"]["substatus"]
         self.assertEquals(len(substatus_file_data), 1)
         self.assertTrue(substatus_file_data[0]["name"] == Constants.CONFIGURE_PATCHING_SUMMARY)
-        self.assertTrue(substatus_file_data[0]["status"] == Constants.STATUS_ERROR.lower())
+        if runtime.vm_cloud_type == Constants.VMCloudType.AZURE:
+            self.assertTrue(substatus_file_data[0]["status"] == Constants.STATUS_ERROR.lower())
+            self.assertTrue(len(json.loads(substatus_file_data[0]["formattedMessage"]["message"])["errors"]["details"]), 1)
+            self.assertEquals(json.loads(substatus_file_data[0]["formattedMessage"]["message"])["errors"]["details"][0]["message"], "Error: Exception('Unsupported older Azure Linux Agent version. To resolve: http://aka.ms/UpdateLinuxAgent')")
```

Comment:
Instead of using the string, try using TELEMETRY_AT_AGENT_NOT_COMPATIBLE_ERROR_MSG from Constants. This will avoid re-work and test failure if the error text is modified in future.

check can be:
self.assertTrue(Constants.TELEMETRY_AT_AGENT_NOT_COMPATIBLE_ERROR_MSG in json.loads(substatus_file_data[0]["formattedMessage"]["message"])["errors"]["details"][0]["message"])